src_kt/java/arcs/data/util/ReferencablePrimitive.kt

```diff
@@ -14,16 +14,94 @@ package arcs.data.util
 import arcs.common.Referencable
 import arcs.common.ReferenceId
 import arcs.util.Base64
+import arcs.util.toBase64Bytes
+import kotlin.reflect.KClass
 
-data class ReferencablePrimitive<T>(val value: T, val valueRepr: String? = null) : Referencable {
+/**
+ * Like the name suggests, this represents a primitive which can be referenced - and thus used by
```

Comment:
"Like the name suggests" lol

...but also, probably not needed. 😛Just: "Represents a primitive which ..."